src/middleware.ts

```diff
@@ -20,8 +21,13 @@ const middleware: NextMiddleware = async (req: NextRequest) => {
     if (user && req.nextUrl.pathname === "/login") {
         return NextResponse.redirect(new URL("/clients", req.url));
     }
-
-    return res;
+    const userRole = user?.app_metadata.role;
+    const blockedPages = roleToHiddenPages[userRole] ?? [];
+    if (blockedPages.includes(req.nextUrl.pathname)) {
+        const url = req.nextUrl;
+        url.pathname = "/404";
```

Comment:
is this actually the 404 page or just an 'invalid url' which then it redirected to the 404?

Reply:
Redirects to the 404 page, with lists still in the URL